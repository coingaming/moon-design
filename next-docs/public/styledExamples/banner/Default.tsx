import { Banner } from '@heathmont/moon-core';

const Example = () => (
  <Banner title="Title here" description="Lorem ipsum dolor..." />
);

export default Example;
