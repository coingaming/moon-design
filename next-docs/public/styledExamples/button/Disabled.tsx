import { Button } from '@heathmont/moon-core';

const Example = () => <Button disabled>Disabled</Button>;

export default Example;
