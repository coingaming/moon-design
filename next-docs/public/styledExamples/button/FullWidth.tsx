import { Button } from '@heathmont/moon-core';
import { GenericSettings } from '@heathmont/moon-icons';

const Example = () => (
  <Button iconRight={<GenericSettings />} fullWidth>
    Full width
  </Button>
);

export default Example;
