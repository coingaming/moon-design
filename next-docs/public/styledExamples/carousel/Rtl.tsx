import { Carousel } from '@heathmont/moon-components';
import { rem } from '@heathmont/moon-utils';
import styled from 'styled-components';

const ExampleContent = styled.div({
  width: rem(320),
  height: rem(192),
  borderRadius: "0.5rem",
  background: 'darkgray',
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
});

const Example = () => {
  const items = Array.from({ length: 25 }, (_, index) => index);
  return (
    <Carousel
      isRtl={true}
      scrollTo={5}
      items={items.map((_, index) => (
        <ExampleContent>{index}</ExampleContent>
      ))}
    />
  );
};

export default Example;
