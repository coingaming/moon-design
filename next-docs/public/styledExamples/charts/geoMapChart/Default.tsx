// @ts-nocheck
import React from 'react';
import { Filter, GeoMapChart } from '@heathmont/moon-charts';
import { SingleItemSelect } from '@heathmont/moon-components';

const Example = () => {
  const data = [
    { label: 'Estonia', code: 'EST', value: 120 },
    { label: 'Russia', code: 'RUS', value: 100 },
    { label: 'Belarus', code: 'BLR', value: 80 },
    { label: 'Argentina', code: 'ARG', value: 60 },
    { label: 'USA', code: 'USA', value: 40 },
  ];

  const filter = (
    <Filter title="by GGR">
      {() => (
        <SingleItemSelect
          onChange={() => {}}
          value="ggr"
          options={[
            { label: 'Active players', value: 'actives' },
            { label: 'GGR', value: 'ggr' },
            { label: 'Bets', value: 'bets' },
          ]}
        />
      )}
    </Filter>
  );

  return (
    <GeoMapChart
      title="Top Countries"
      data={data}
      filter={filter}
      hasUpdates={true}
      onUpdate={() => {}}
      onShare={() => {}}
      onExpand={() => {}}
    />
  );
};

export default Example;
