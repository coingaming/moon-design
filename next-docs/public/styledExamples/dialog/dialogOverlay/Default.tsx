import React from 'react';
import { DialogOverlay } from '@heathmont/moon-components';
import { Button } from '@heathmont/moon-core';

const Example = () => {
  const [showDialog, setShowDialog] = React.useState(false);

  return (
    <>
      <Button variant="tertiary" onClick={() => setShowDialog(true)}>
        Open Dialog
      </Button>
      <DialogOverlay
        isOpen={showDialog}
        onDismiss={() => setShowDialog(false)}
      />
    </>
  );
};

export default Example;
