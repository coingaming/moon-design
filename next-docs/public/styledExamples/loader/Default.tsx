import React from 'react';
import { Loader } from '@heathmont/moon-core';

const Example = () => <Loader />;

export default Example;
