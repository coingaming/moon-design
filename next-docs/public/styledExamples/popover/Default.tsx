import React, { useCallback, useEffect, useState } from 'react';
import { Popover } from '@heathmont/moon-popover';
import { Footer, Heading, Text, Button } from '@heathmont/moon-core';

const Example = () => {
  const [open, setOpen] = useState(false);

  const HandlerOpen = useCallback(
    (e: any) => {
      e.preventDefault();
      setOpen(false);
    },
    [open]
  );

  return (
    <Popover
      padding={0}
      placement="bottom"
      isOpen={open}
      onVisibilityChange={(status) => {
        setOpen(status);
        console.log('status: ', status);
      }}
      content={
        <div>
          <div className="p-8 space-y-4">
            <Heading size={20}>Agents</Heading>
            <Text size={14}>
              Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
              eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut
              enim ad minim veniam, quis nostrud exercitation ullamco laboris
              nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in
              reprehenderit in voluptate velit esse cillum dolore eu fugiat
              nulla pariatur.
            </Text>
          </div>

          <Footer
            primButton={
              <Button fullWidth size="small" onClick={HandlerOpen}>
                OK
              </Button>
            }
            secButton={
              <Button
                fullWidth
                variant="secondary"
                size="small"
                onClick={HandlerOpen}
              >
                Cansel
              </Button>
            }
            tertButton={
              <Button
                fullWidth
                variant="tertiary"
                size="small"
                onClick={HandlerOpen}
              >
                Decline
              </Button>
            }
            isDivider
            size="small"
          />
        </div>
      }
    >
      <Button>Toggle Popover</Button>
    </Popover>
  );
};

export default Example;
