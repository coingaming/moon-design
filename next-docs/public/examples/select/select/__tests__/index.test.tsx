/**
 * @jest-environment jsdom
 */

import React from 'react';
import { moonDesignLight, ThemeProvider } from '@heathmont/moon-themes';
import renderer from 'react-test-renderer';
import Default from '../Default';
import Label from '../Label';
import Rtl from '../Rtl';
import FlexSpace from '../FlexSpace';
import AdjustedInputSize from '../AdjustedInputSize';
import Disabled from '../Disabled';
import FullWidth from '../FullWidth';

const renderWithTheme = (component: JSX.Element) => (
  <ThemeProvider theme={moonDesignLight}>{component}</ThemeProvider>
);

const renderWithThemeRTL = (component: JSX.Element) => (
  <ThemeProvider theme={moonDesignLight}>
    <div dir="rtl">{component}</div>
  </ThemeProvider>
);

describe('Reel', () => {
  it('renders', () => {
    const testRenderer = renderer.create(renderWithTheme(<Default />));
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });

  it('renders RTL', () => {
    const testRenderer = renderer.create(renderWithThemeRTL(<Rtl />));
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });

  it('renders with label', () => {
    const testRenderer = renderer.create(renderWithTheme(<Label />));
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });

  it('renders with flex evenly spaced by default', () => {
    const testRenderer = renderer.create(renderWithTheme(<FlexSpace />));
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });

  it('renders with flex with adjusted input size', () => {
    const testRenderer = renderer.create(
      renderWithTheme(<AdjustedInputSize />)
    );
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });

  it('renders as disabled', () => {
    const testRenderer = renderer.create(renderWithTheme(<Disabled />));
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });

  it('renders as full-width', () => {
    const testRenderer = renderer.create(renderWithTheme(<FullWidth />));
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
});
