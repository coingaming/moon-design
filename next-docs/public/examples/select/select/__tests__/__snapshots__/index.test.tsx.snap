// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reel renders 1`] = `
<select
  className="sc-eIvVYG sc-fESQEH iBFsit cnzzWj"
>
  <option
    value=""
  >
    Select an option
  </option>
  <option
    value="never-gonna"
  >
    Never gonna
  </option>
  <option
    value="give-you-up"
  >
    give you up
  </option>
  <option
    value="let-you-down"
  >
    let you down
  </option>
</select>
`;

exports[`Reel renders RTL 1`] = `
<div
  dir="rtl"
>
  <select
    className="sc-eIvVYG sc-fESQEH iBFsit hiJwjk"
    dir="rtl"
  >
    <option
      value=""
    >
      Select an option
    </option>
    <option
      value="never-gonna"
    >
      Never gonna
    </option>
    <option
      value="give-you-up"
    >
      give you up
    </option>
    <option
      value="let-you-down"
    >
      let you down
    </option>
  </select>
</div>
`;

exports[`Reel renders as disabled 1`] = `
<select
  className="sc-eIvVYG sc-fESQEH iBFsit cnzzWj"
  disabled={true}
>
  <option
    value=""
  >
    Select an option
  </option>
  <option
    value="never-gonna"
  >
    Never gonna
  </option>
  <option
    value="give-you-up"
  >
    give you up
  </option>
  <option
    value="let-you-down"
  >
    let you down
  </option>
</select>
`;

exports[`Reel renders as full-width 1`] = `
<select
  className="sc-eIvVYG sc-fESQEH iBFsit gKcQNE"
>
  <option
    value=""
  >
    Select an option
  </option>
  <option
    value="never-gonna"
  >
    Never gonna
  </option>
  <option
    value="give-you-up"
  >
    give you up
  </option>
  <option
    value="let-you-down"
  >
    let you down
  </option>
</select>
`;

exports[`Reel renders with flex evenly spaced by default 1`] = `
<select
  className="sc-eIvVYG sc-fESQEH iBFsit cnzzWj"
>
  <option
    value=""
  >
    Select an option
  </option>
  <option
    value="never-gonna"
  >
    Never gonna
  </option>
  <option
    value="give-you-up"
  >
    give you up
  </option>
  <option
    value="let-you-down"
  >
    let you down
  </option>
</select>
`;

exports[`Reel renders with flex with adjusted input size 1`] = `
<select
  className="sc-eIvVYG sc-fESQEH iBFsit cnzzWj"
>
  <option
    value=""
  >
    Select an option
  </option>
  <option
    value="never-gonna"
  >
    Never gonna
  </option>
  <option
    value="give-you-up"
  >
    give you up
  </option>
  <option
    value="let-you-down"
  >
    let you down
  </option>
</select>
`;

exports[`Reel renders with label 1`] = `
<label
  className="sc-gouySG"
>
  <span
    className="sc-kgSa-dv fjlUwV"
  >
    Label Content
  </span>
  <select
    className="sc-eIvVYG sc-fESQEH iBFsit cnzzWj"
  >
    <option
      value=""
    >
      Select an option
    </option>
    <option
      value="never-gonna"
    >
      Never gonna
    </option>
    <option
      value="give-you-up"
    >
      give you up
    </option>
    <option
      value="let-you-down"
    >
      let you down
    </option>
  </select>
</label>
`;
